### DISCUSSION TITLE
Help: Routing Deluge through GluetunVPN

### DISCUSSION THREAD

**Discussion by Aranethon** [qdm12/gluetun#963] [Ideas]
I've looked all over and tried a number of solutions, but unfortunately I'm still having issues getting Deluge to work while the container is routed through Gluetun.

I'm using Unraid.
My VPN is Surfshark, using OpenVPN.
I connect my Deluge container to my Gluetun container by adding Deluge's ports to Gluetun then setting up Deluge's network to `none`, and its extra parameters to `--net=container:GluetunVPN`.
Doing this I am able to access the WebUI, and the VPN IP is correct, but no seeds will connect. The status returns `Error: Host not found (authoritative)`

I've also attempted to add the ports to `FIREWALL_INPUT_PORTS:` and `FIREWALL_VPN_INPUT_PORTS:` both independently and together with the same results.
Lastly I turned off the Firewall completely, and the problem still remained.

What else can I do to try and fix this issue?

**Comment by qdm12** [comment]
Hello there, maybe setting the Deluge network to none is the problem?
I'm not super familiar with the Unraid setting though.
You just have to publish the port of the web ui on gluetun to access it, there is no need to fiddle with `FIREWALL_INPUT_PORTS` or `FIREWALL_VPN_INPUT_PORTS` really.

If that doesn't work, isn't there a way to use `docker-compose.yml` files in Unraid?

**Reply by Aranethon** [reply]
> maybe setting the Deluge network to none is the problem?

It's a required step for the extra parameter to work, else there would be port conflicts with the containers. It's how you route all a container's traffic through another in Unraid.

> If that doesn't work, isn't there a way to use `docker-compose.yml` files in Unraid?

I believe there is through the use of some unofficial plugins.

Transmission also had a similar issue.
In the end I found that using qBittorrent worked for some reason, although a good number of trackers on any given torrent return `Skipping tracker announce (unreachable)`.
I'm unsure if this is a result of the previous issue, or if it's due to Gluetun, or my particular VPN provider/endpoint.

**Reply by qdm12** [reply]
Maybe try `docker exec -it deluge bash` and try pinging/nslookup in the container to see if connectivity is completely down or if it's just i. e. a dns issue.
